$NetBSD: patch-ba,v 1.6 2026/05/19 18:38:11 spz Exp $

deal with ctype functions faulting on invalid input on NetBSD 11

add more OS variance handling

--- term.c.orig	2005-06-29 21:40:27.000000000 +0000
+++ term.c
@@ -14,7 +14,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <string.h>
-#include <ctype.h>
+#include "ctypeconv.h"
 #include "config.h"
 #include "global.h"
 #include "execute.h"
@@ -25,7 +25,7 @@
 #include "regexp.h"
 #include "nn_term.h"
 
-#if !defined(__FreeBSD__) && !(__NetBSD__) && !defined(NeXT)
+#if !defined(__FreeBSD__) && !(__NetBSD__) && !defined(NeXT) && !defined(__APPLE__) && !defined(__DragonFly__)
 #include <stropts.h>
 #else
 #include <sys/ioctl.h>
@@ -60,7 +60,7 @@ extern int      s_resized;
 #else
 #define USE_TERMCAP
 
-#if !defined(SUNOS4) && !defined(NeXT)
+#if !defined(SUNOS4) && !defined(NeXT) && !defined(__sgi)
 #include <termcap.h>
 #endif				/* SUNOS4 */
 
@@ -74,9 +74,9 @@ extern int      s_resized;
 
 #else
 
-#ifndef __FreeBSD__
+#ifndef HAVE_TERMIOS_H
 #include <sgtty.h>
-#endif				/* __FreeBSD__ */
+#endif				/* HAVE_TERMIOS_H */
 
 #endif
 
@@ -456,6 +456,12 @@ siginterrupt(signo, on)
 #endif				/* FAKE_INTERRUPT */
 
 static unsigned sp_table[] = {
+#ifdef B115200
+    B115200, 11520,
+#endif
+#ifdef B57600
+    B57600, 5760,
+#endif
     B9600, 960,
 
 #ifdef B19200
